src/wadray.cairo

```diff
@@ -255,13 +255,11 @@ pub impl RayDivAssign of DivAssign<Ray, Ray> {
 
 
 // Conversions
-pub impl WadTryIntoRay of TryInto<Wad, Ray> {
-    fn try_into(self: Wad) -> Option::<Ray> {
-        if (self.val <= MAX_CONVERTIBLE_WAD) {
-            Option::Some(Ray { val: self.val * DIFF })
-        } else {
-            Option::None
-        }
+pub fn wad_to_ray(x: Wad) -> Option<Ray> {
```

Comment:
As they say, a bit of a pickle, but this seems to be the best solution.

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExYjc5OWNiaWJ1ejFpY2hqNWptNmpwaWttbW42anFoeWY2dWo4MWw4byZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/xUOrwd4k6IK8baHImk/giphy.gif)